src/player.h

```diff
@@ -1602,6 +1615,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         /** Stamp of skills. @ref learned_recipes are valid only with this set of skills. */
         mutable decltype( _skills ) valid_autolearn_skills;
 
+        /** smart pointer to targeting data stored for aiming the player's weapon across turns. */
+        std::shared_ptr<targeting_data> tdata;
```

Comment:
`shared_ptr` is also copyable. `unique_ptr` isn't, and it breaks calls to `player`'s copy/assignment constructors. 

The benefit of `unique_ptr` is that it ensures an object isn't shared/owned by anything else, but we don't need that guarantee here. Given the current class architecture, and since we are copying `player` objects in various places throughout the code, allowing multiple instances of `player` to (temporarily) share a `tdata` value is appropriate and desirable.

If `player` really wants to use `unique_ptr` for this and still be copyable, it needs to implement a constructor that can make a deep copy of `tdata`. It would be a lot of work just to get `tdata` to pretend it's a copyable smart pointer, which is what `shared_ptr` already is.

The alternative is deleting the copy/assignment ctors from `player` and `npc` and changing copy operations throughout the source, which would be quite a task as there are multiple functions throughout the source that rely on `player` copies.